### PR TITLE
Add norfork data

### LINKED ISSUE #232: Bring in Norfolk_* temperature data
Norfolk_* data set consists of ten files for two locations at the ***Norfork*** Reservoir Dam in northern Arkansas: the dam site and highway bridge 62. Data for both sites was collected on a monthly basis from 2016-2020 and was received from AR Fish and Game Commission. File prefixes are as follows:
* Dam site: Norfolk_DAM-YYYY.xlsx
* Bridge site: Norfolk_62BRG-YYYY.xlsx

It should be noted that a google search did not return any results for "Norfolk Reservoir Arkansas". The only results returned were for ***Norfork***. To account for this typo, the parser will be called `parse_norfork_data`, but the data files will not be renamed. Data and parser will be matched in [`parsing_task_fxns.R`](https://github.com/USGS-R/lake-temperature-model-prep/blob/main/7a_temp_coop_munge/src/parsing_task_fxns.R).

Based on a spot check of  the data in the above-mentioned files, the two additional files labeled as Norfolk in the [`queue`](https://drive.google.com/drive/folders/1jGdWewg7uPaMvUwLMWtvfTEwSyYzHQDi) folder appear to contain the same data (`NorfolkReservoir_hwy62_AR_monthlyTempDO_2016-2020.xlsx` and `NorfolkReservoir_dam_AR_monthlyTempDO_2016-2020`). These two files will be ignored.

Within the ten files of interest, the data appears relatively uniform. The first two excel sheets are typically depth/temperature profiles and aggregated data. The remainder of the sheets in each workbook are formatted in `day-month-year` format (e.g. `6-Aug-2016` or `29-July-2016`) and correspond to individual sampling events. Sampling events include the following data:
* Timestamp - includes date and time, timezone extracted from data is UTC
* Barometer (kPA)
* Dissolved oxygen (mg/L)
* Temperature (C)
* Unit ID
* Depth (ft) - manually calculated and referenced from the surface. This column is often not labeled.

### LINKED ISSUE #235: Helper functions for unit conversions
We are seeing unit conversions happen in multiple functions across the pipeline. It might be good to have a single set of helper functions to do unit conversions that can be leveraged across the pipeline, so that they are all using the same decimal places and process. 

See short discussion [here](https://github.com/USGS-R/lake-temperature-model-prep/pull/218#discussion_r748768535).

### PR DESCRIPTION
**Pull request by padilla410**
This pull request completes the following:
* Adds `Norfork_*` data to `all_coop_dat_linked.feather` (closes #232)
* Creates a spatial crosswalk for the `Norfork_*` 
* Creates a helper function to ensure consistent unit conversion from ft to m (closes #235)
* Successfully runs the pipeline through `scmake("8_viz")`

### Adding data to `all_coop_dat_linked.feather`
Run the code below to verify that the Norfork data sets were added to the `all_coop_dat_linked.feather`
```r
# verification that Norfork data is in munged data set
dat_feather <- read_feather(sc_retrieve('7a_temp_coop_munge/out/all_coop_dat_linked.feather.ind'))

filt_dat_norfork <- dat_feather %>% 
  filter(str_detect(source, "Norfolk_")) %>% 
  group_by(source) %>% 
  summarise(count = n())
````

### Adding the spatial cross walk
The spatial crosswalk for this dataset was created based on descriptive information provided from `6_temp_coop_fetch/explain/Norkfolk_62BRG-XXXX.docx` and `6_temp_coop_fetch/explain/Norkfolk_DAM-XXXX.docx`. Originally, I had manually assigned an `nhdhr` id to the data set  because it only represents one location (a la PR #218), but that resulted in the omission of the data from `all_coop_dat_linked.feather`. the Norfork data was omitted because `crosswalk_coop_dat()` does not have a way to convert
an `id` field (with an `nhdhr` id) to a `site_id` field without a crosswalk. In working to debug this issue, I also found a few data sets that likely have the same problem. This has been documented in issue #267.

When completing this step I used the directions in the repo `README` and in Hayley's Navico PR #210. 

### Adding the helper function
This function was added to `convert_ft_to_m()` was added to `lib/src/generic_utils.R` to ensure
consistent unit conversions between data sets.

### Successful run of `scmake("8_viz")`
Snapshot of `8_viz/out/lakes_summary_fig.html`:
image

Note: there are no additional lakes listed in this figure (when compared to PR #263) because both the Norfork and Navico datasets contain data for Norfork Lake in Northwest Arkansas.

```r
# verification that Norfork data is in munged data set
dat_feather <- read_feather(sc_retrieve('7a_temp_coop_munge/out/all_coop_dat_linked.feather.ind'))

count_by_dataset <- dat_feather %>% 
  filter(str_detect(source, "Norfolk_") | str_detect(source, "Waterbody_")) %>% 
  group_by(source) %>% 
  summarise(count = n())

````

### CODE REVIEW

File: 6_temp_coop_fetch/out/coop_all_files.rds.tind [hunk 1]
```diff
@@ -1,3 +1,3 @@
-hash: 8f2a204993aa7765bb50e101046d6c98
-freshened: 2021-12-20 09:49:26 -0500
+hash: e5d3bf3717099c1c1c924ce6d115b32d
```
Comment (jread-usgs) [inline 6_temp_coop_fetch/out/coop_all_files.rds.tind:1]: :+1:

**Review by jread-usgs — APPROVED** [review]
Nice work. I built this and it worked as expected. Crosswalk looks good

File: 7a_temp_coop_munge/src/data_parsers/parse_norfork_files.R [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+      time = format(Timestamp, "%H:%M"),
+      Timezone = format(Timestamp, "%Z"),
+      depth = convert_ft_to_m(`...10`) * -1, # depth is referenced from water surface
+      temp = `Temperature (C)`,
+      # One lake with two sites. Manually set from
+      # https://github.com/USGS-R/lake-temperature-model-prep/blob/0c037ddce5bdab7a7eb3a6ad07c0837972bddcd1/4_params_munge.yml#L63
+      Norfork_ID = paste("Norfork_", `Unit ID`, sep = ''),
+      site = Site
+    ) %>%
+    dplyr::select(DateTime, time, Timezone, depth, temp, Norfork_ID, site)
```
[the remark this answers is not in the document]
Comment (padilla410) [reply 7a_temp_coop_munge/src/data_parsers/parse_norfork_files.R:24]: This is a good point. According to the [doc](https://docs.google.com/document/d/1Z1fg8ng9IWHS9oiQTxsDvpGBlbgz2bMZ5rWNxfRiHC8/edit#heading=h.5ajsuzlpviwv), `site` is an optional variable, but I should only use it when there are multiple sites per lake. I'm going to drop `site` from the parser. I will add one more commit and then merge this PR.

[the remark this answers is not in the document]
Comment (padilla410) [reply 7a_temp_coop_munge/src/data_parsers/parse_norfork_files.R:24]: Update! I am going to walk back the above comment. The Norfork Reservoir does, in fact, have two sampling locations on one waterbody. One location is at the dam and the other is off of the bridge at Rte 62 (~right around the middle of the reservoir). 

I will not be removing `site` from the data parser, but I will be submitting one more commit to make sure the hashes are up to date (discovered this when I rebuilt the target this morning).

[the remark this answers is not in the document]
Comment (jread-usgs) [reply 7a_temp_coop_munge/src/data_parsers/parse_norfork_files.R:24]: Good plan!

**Pull request merged**